src/main/java/org/wildfly/security/authz/jacc/ElytronPolicyConfiguration.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.authz.jacc;
+
+import javax.security.jacc.PolicyConfiguration;
+import javax.security.jacc.PolicyContextException;
+import java.security.Permission;
+import java.security.PermissionCollection;
+import java.security.Permissions;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+import static org.wildfly.security._private.ElytronMessages.log;
+
+/**
+ * {@link javax.security.jacc.PolicyConfiguration} implementation.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ * @see org.wildfly.security.authz.jacc.ElytronPolicyConfigurationFactory
+ */
+class ElytronPolicyConfiguration implements PolicyConfiguration {
```

Comment:
I've the same understanding. I didn't consider any type of concurrency control because PB's DelegatingPolicy does not provide anything around that today. I'm considering that if it is working without any issue today, I should follow the same approach and avoid synchronizing things on ```addToUncheckedPolicy```, ```addToUncheckedPolicy```, ```removeRole``` and others.

Also, since a policy config is specific to a deployment and created only once during the deployment phase, I'm considering that we really don't need to deal with concurrency controls. 

Or do you think we should consider concurrency and do something different than PB in this sense ?